### PR TITLE
Some Soundcache and bot updates

### PR DESCRIPTION
**Pull request by BenCat07**
Exactly what it says

### CODE REVIEW

File: include/core/offsets.hpp [hunk 1]
```diff
@@ -196,4 +196,16 @@ struct offsets
     {
         return PlatformOffset(128, undefined, undefined);
     }
+    static constexpr uint32_t EmitSound1()
```
Comment (TotallyNotElite) [inline include/core/offsets.hpp]: Are these still needed?

File: include/hooks/HookedMethods.hpp [hunk 2]
```diff
@@ -86,6 +86,11 @@ DECLARE_HOOKED_METHOD(RandomInt, int, IUniformRandomStream *, int, int);
 #if ENABLE_VISUALS || ENABLE_TEXTMODE
 DECLARE_HOOKED_METHOD(DrawModelExecute, void, IVModelRender *, const DrawModelState_t &, const ModelRenderInfo_t &, matrix3x4_t *);
 #endif
+// g_ISoundEngine
+DECLARE_HOOKED_METHOD(EmitSound1, void, void *, IRecipientFilter &, int, int, const char *, float, float, int, int, int, const Vector *, const Vector *, CUtlVector<Vector> *, bool, float, int);
```
Comment (TotallyNotElite) [inline include/hooks/HookedMethods.hpp]: Same issue here (is 1 and 3 still needed)

File: src/hack.cpp [hunk 3]
```diff
@@ -336,6 +336,12 @@ free(logname);*/
     hooks::steamfriends.HookMethod(HOOK_ARGS(GetFriendPersonaName));
     hooks::steamfriends.Apply();
 
+    hooks::soundclient.Set(g_ISoundEngine);
+    hooks::soundclient.HookMethod(HOOK_ARGS(EmitSound1));
```
Comment (TotallyNotElite) [inline src/hack.cpp]: 1 and 3 still needed?

File: src/hooks/EmitSound.cpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#include "common.hpp"
+#include "HookedMethods.hpp"
+#include "soundcache.hpp"
+
+namespace hooked_methods
+{
+DEFINE_HOOKED_METHOD(EmitSound1, void, void *_this, IRecipientFilter &filter, int iEntIndex, int iChannel, const char *pSample, float flVolume, float flAttenuation, int iFlags, int iPitch, int iSpecialDSP, const Vector *pOrigin, const Vector *pDirection, CUtlVector<Vector> *pUtlVecOrigins, bool bUpdatePositions, float soundtime, int speakerentity)
```
Comment (TotallyNotElite) [inline src/hooks/EmitSound.cpp]: 1 and 3 still needed?

File: src/hooks/nographics.cpp [hunk 5]
```diff
@@ -286,4 +288,23 @@ static InitRoutine nullifiy_textmode2([]() {
...
     });
+#if ENABLE_TEXTMODE
+    auto addr = gSignatures.GetEngineSignature("E8 ? ? ? ? C7 04 24 ? ? ? ? E8 ? ? ? ? C7 04 24 ? ? ? ? E8 ? ? ? ? E8 ? ? ? ?");
+    addr      = e8call_direct(addr);
+    static BytePatch patch(addr, { 0x31, 0xc0, 0xc3 });
+    patch.Patch();
+    EC::Register(
+        EC::Paint,
+        []() {
+            if (!signon_timer.test_and_set(5000))
```
Comment (TotallyNotElite) [inline src/hooks/nographics.cpp]: Janky hack

File: include/core/offsets.hpp (hunk 1 above)
Comment (BenCat07) [reply include/core/offsets.hpp]: Just in Case

File: src/hack.cpp (hunk 3 above)
Comment (BenCat07) [reply src/hack.cpp]: Just in Case

File: src/hooks/nographics.cpp (hunk 5 above)
Comment (BenCat07) [reply src/hooks/nographics.cpp]: Tested, works flawlessly, but yes it's a janky hack (we don't really have a choice here)

File: src/hooks/EmitSound.cpp (hunk 4 above)
Comment (BenCat07) [reply src/hooks/EmitSound.cpp]: Just in Case

File: include/hooks/HookedMethods.hpp (hunk 2 above)
Comment (BenCat07) [reply include/hooks/HookedMethods.hpp]: Just in Case

File: src/hooks/nographics.cpp (hunk 5 above)
Comment (TotallyNotElite) [reply src/hooks/nographics.cpp]: Have you tried just not adding that there?

Comment (BenCat07) [reply src/hooks/nographics.cpp]: it won't ever finish loading a map if i do

File: src/votelogger.cpp [hunk 6]
```diff
@@ -67,7 +68,8 @@ void dispatchUserMessage(bf_read &buffer, int type)
     switch (type)
     {
     case 45:
-        // Call Vote Failed
+        // Call Vote Failed, Refresh vote timer for catbot so it can try again
+        hacks::shared::catbot::timer_votekicks.last -= std::chrono::seconds(4);
```
Comment (TotallyNotElite) [inline src/votelogger.cpp]: You should verify if this is actually your own votekick or not.

Comment (TotallyNotElite) [reply src/votelogger.cpp]: image

**Pull request merged**